tensornetwork/linalg/initialization.py

```diff
@@ -105,6 +107,51 @@ def ones(shape: Sequence[int],
   the_tensor = initialize_tensor("ones", shape, backend=backend, dtype=dtype)
   return the_tensor
 
+def ones_like(input: Union[np.ndarray,Tensor],
+         dtype: Optional[Type[np.number]] = None,
+         backend: Optional[Union[Text, AbstractBackend]] = None) -> Tensor:
+  """Return a Tensor shape full of ones the same shape as input
+  Args:
+    input : Object to recieve shape from
+     dtype (optional) : dtype of object
+     backend(optional): The backend or its name."""
+  if backend is None:
+    backend = backend_contextmanager.get_default_backend()
+  elif isinstance(backend, str):
+    backend = backend_contextmanager.backend_factory.get_backend(backend)
+  else:
+    backend = backend
+  if isinstance(input, Tensor): # encase input of type Tensor, create Tensor normally
+    the_tensor = initialize_tensor("ones", input.shape, backend=input.backend, dtype=input.dtype)
+    return the_tensor
+  if isinstance(input, np.ndarray): # encase input of type np.ndarray, convert to Tensor
```

Comment:
I think actually a simple "else" would be better here - presently if the input is neither a an ndarray nor a Tensor, this function will silently return None. In addition we want to catch other backend types than ndarray